### PR TITLE
Update getting-started-with-swashbuckle.md

### LINKED ISSUE #31437: Swashbuckle is not supported on .NET 8
### Description

For .NET 8 Swashbuckle is mentioned as an option of working with OpenAPI specs.
However, Swashbuckle have not been updated for the past year, and is incompatible with .NET 8

See issues like https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2707 & https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2725

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?view=aspnetcore-8.0&tabs=visual-studio

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/getting-started-with-swashbuckle.md

### Document ID

59222460-a971-e76e-4fe9-23d4d7d768ee

### Article author

@zuckerthoben

---
[Associated WorkItem - 199402](https://dev.azure.com/msft-skilling/Content/_workitems/edit/199402)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #31437

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/tutorials/getting-started-with-swashbuckle.md](https://github.com/dotnet/AspNetCore.Docs/blob/f18d6e9ea9ece562606727a8b506dbc0ebede8f5/aspnetcore/tutorials/getting-started-with-swashbuckle.md) | [Get started with Swashbuckle and ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?branch=pr-en-us-31441) |

### CODE REVIEW

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ uid: tutorials/get-started-with-swashbuckle
 ---
 # Get started with Swashbuckle and ASP.NET Core
 
-:::moniker range=">= aspnetcore-6.0"
+:::moniker range=">= aspnetcore-6.0 < aspnetcore-8.0"
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md:13]:
```suggestion
:::moniker range=">= aspnetcore-8.0 
Swashbuckle is not supported in current versions of .NET. For a supported alternative, see <xref:tutorials/web-api-help-pages-using-swagger>.

:::moniker range=">= aspnetcore-6.0 < aspnetcore-8.0"
```

**Comment by Rick-Anderson** [comment]
@tdykstra  I'd rather just hide the topic for .NET 8 and higher, but that doesn't work.

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md
Comment (tdykstra) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md:1]: What about changing the file metadata from

monikerRange: '>= aspnetcore-3.1'

to

monikerRange: '>= aspnetcore-3.1 < aspnetcore-8.0'

That would at least get it out of the TOC.

**Review by tdykstra — APPROVED** [review]
One idea for you to consider.

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md
Comment (Rick-Anderson) [reply aspnetcore/tutorials/getting-started-with-swashbuckle.md:1]: > What about changing the file metadata from
> 
> monikerRange: '>= aspnetcore-3.1'
> 
> to
> 
> monikerRange: '>= aspnetcore-3.1 < aspnetcore-8.0'
> 
> That would at least get it out of the TOC.

That was my first try, and at least in the review URL, it still showed up in the TOC.

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 2]
```diff
@@ -10,7 +10,10 @@ uid: tutorials/get-started-with-swashbuckle
 ---
 # Get started with Swashbuckle and ASP.NET Core
 
-:::moniker range=">= aspnetcore-6.0"
+:::moniker range=">= aspnetcore-8.0 
+Swashbuckle is not supported in current versions of .NET. For a supported alternative, see <xref:tutorials/web-api-help-pages-using-swagger>.
+
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md:15]:
```suggestion
<!--
:::moniker range=">= aspnetcore-8.0 
Swashbuckle is not supported in current versions of .NET. For a supported alternative, see <xref:tutorials/web-api-help-pages-using-swagger>.
-->
```

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md
Comment (Rick-Anderson) [reply aspnetcore/tutorials/getting-started-with-swashbuckle.md:1]: @tdykstra  take a look, it still shows up in the TOC.

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 3]
```diff
@@ -10,7 +10,12 @@ uid: tutorials/get-started-with-swashbuckle
 ---
 # Get started with Swashbuckle and ASP.NET Core
 
-:::moniker range=">= aspnetcore-6.0"
+<!--
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md:13]:
```suggestion
```

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 4]
```diff
@@ -10,7 +10,12 @@ uid: tutorials/get-started-with-swashbuckle
...
+:::moniker range=">= aspnetcore-8.0 
+Swashbuckle is not supported in current versions of .NET. For a supported alternative, see <xref:tutorials/web-api-help-pages-using-swagger>.
+:::moniker-end
+-->
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md:17]:
```suggestion
```

**Comment by tdykstra** [comment]
With "< aspnetcore-8.0" in the file metadata you don't need the < aspnetcore-8.0 block in the text.  It always switches to 7.0 when you try to go to 8.0.

**Comment by Rick-Anderson** [comment]
It's still not a great user experience if you land on that page so I added the warning back.

**Comment by Rick-Anderson** [comment]
Can you fix the file? Just edit it.

**Comment by tdykstra** [comment]
OK I think it's ready to go now:
* If you try to go to the page with 8.0+ selected, it automatically kicks you back to 7.0.
* All versions warn that it's not supported in 8.0+

**Comment by Rick-Anderson** [comment]
> All versions warn that it's not supported in 8.0+
That's good. Even if you're still on 6 or 7, better to get a warning.

**Pull request merged**